### PR TITLE
Rotating Grids that kind of works

### PR DESCRIPTION
**Pull request by ghost**
Does a lot of things to make rotating grids work, including sprite rotation and improving vector math. It's very rough and the calculation to determine which gridchunks to render is broken. I give up on that one, maybe I'm retarded. Ask if you need any details on it.

### CODE REVIEW

File: Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs [hunk 1]
```diff
@@ -61,9 +65,11 @@ private void _drawGrids(Box2 worldBounds)
                     continue;
                 }
 
-                var model = Matrix3.Identity;
+                Matrix3 model = Matrix3.Identity;
```
Comment (PJB3005) [inline Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs:68]: Why remove these `var` declarations :thinking:

File: Robust.Client/Graphics/ClientEye/Eye.cs [hunk 2]
```diff
@@ -14,7 +14,10 @@ public class Eye : IEye
 
         /// <inheritdoc />
         public virtual MapCoordinates Position { get; internal set; }
-        
+
+        /// <inheritdoc />
+        public Angle Rotation { get; set; }
```
Comment (PJB3005) [inline Robust.Client/Graphics/ClientEye/Eye.cs:19]: Didn't get around to using this?

File: Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs (hunk 1 above)
Comment (ghost) [reply Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs:68]: It's more readable. I was also probably planning to change it but went against it.

File: Robust.Client/Graphics/ClientEye/Eye.cs (hunk 2 above)
Comment (ghost) [reply Robust.Client/Graphics/ClientEye/Eye.cs:19]: Nope, this caused an absolute shitton of issues. Grid rotation is already buggy enough. Maybe in the future.

**Comment by ghost** [comment]
And to add on, the normal rotation (0 deg) works fine. When a grid is rotated the calculations for determining what tiles to render is borked. I kind of gave up on it after getting frustrated with Box2Rotated.

File: Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs (hunk 1 above)
Comment (PJB3005) [reply Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs:68]: I disagree. It's completely obvious that this variable is a `Matrix3`. No need to write that again.

Same goes for the changes you made like this in `Clyde.Rendering.cs`.

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs [hunk 3]
```diff
@@ -996,15 +996,13 @@ internal void OpenGLRender(DrawingHandleWorld drawingHandle, bool useWorldTransf
             if (useWorldTransform)
             {
                 var angle = Rotation;
-                if (Directional)
-                {
-                    angle -= Owner.Transform.WorldRotation;
-                }
-                else
+                if (!Directional)
```
Comment (PJB3005) [inline Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs:999]: Are these changes so sprites ignore the rotation of their parent, so they rotate with the grid?

If so, this is gonna break for more complex multi-level layouts (person in chair).

Might be better to just implement the angle snapping idea.

File: Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs (hunk 1 above)
Comment (PJB3005) [inline Robust.Client/Graphics/Clyde/Clyde.GridRendering.cs:68]: Looking at this again, you could probably replace this manual matrix calculation by getting the world matrix from the grid's entity.

**Pull request closed without merging**